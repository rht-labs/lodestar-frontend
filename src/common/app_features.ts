// All roles defined by the backend to be used in the frontend should be enumerated in this object.

export const APP_FEATURES = {
  admin: 'admin',
  writer: 'writer',
  reader: 'reader',
  notifications: 'notifications',
  engagementCardIcons: 'engagementCardIcons',
  resetUser: 'resetUser',
};

export type AppFeature = keyof typeof APP_FEATURES;
