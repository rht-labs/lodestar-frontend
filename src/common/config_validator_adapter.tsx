import { ValidatorFactory } from '../schemas/validators';
import { EngagementFormConfig } from '../schemas/engagement_config';

export const getValidatorsFromFormOptions = (
  formOptions: EngagementFormConfig = {}
) =>
  Object.keys(formOptions || {}).reduce(
    (previousSchemaGroups, currentGroupKey) => {
      return {
        ...previousSchemaGroups,
        ...Object.keys(formOptions[currentGroupKey] ?? {}).reduce(
          (previousGroupFields, currentFieldKey) => ({
            ...previousGroupFields,
            [currentFieldKey]: (
              formOptions?.[currentGroupKey]?.[currentFieldKey]?.validators ||
              []
            ).map(ValidatorFactory),
          }),
          {}
        ),
      };
    },
    {}
  );
