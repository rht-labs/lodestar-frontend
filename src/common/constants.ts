export const APPLICATION_CONTENT_CONTAINER_ID = 'main-application-content';
