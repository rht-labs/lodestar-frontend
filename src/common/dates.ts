import { format as formatDate, utcToZonedTime } from 'date-fns-tz';
export const formatUtcDate = (date?: Date): string | undefined => {
  if (!date) {
    return '';
  }
  const dateString = formatDate(utcToZonedTime(date, 'UTC'), 'MMM dd, yyyy', {
    timeZone: 'UTC',
  });
  return dateString;
};
