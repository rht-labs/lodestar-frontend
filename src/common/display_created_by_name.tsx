import React from 'react';

export function DisplayCreatedByName({
  userFromServer,
  lastUpdatedBy,
}: {
  userFromServer?: string;
  lastUpdatedBy?: string;
}) {
  return <>{!!userFromServer ? userFromServer : lastUpdatedBy}</>;
}
