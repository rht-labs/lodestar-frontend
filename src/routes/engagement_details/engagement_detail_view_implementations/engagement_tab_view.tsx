import React from 'react';
import { EngagementViewProps } from '..';
import { Tabs, Tab } from '@patternfly/react-core';
import { BasicInformation } from '../form_views/01_basic_information';
import { useEngagements } from '../../../context/engagement_context/engagement_hook';
import { PointOfContact } from '../form_views/02_point_of_contact';
import { ClusterUsers } from '../form_views/04_cluster_users';
import { ClusterInformation } from '../form_views/03_cluster_information';
import { EditPaneWrapper } from '../../../components/edit_pane_wrapper/edit_pane_wrapper';
import { useLocation, useHistory } from 'react-router';
import { EngagementOverview } from '../tab_views/overview';
interface EngagementTabViewProps extends EngagementViewProps {}

enum TabNames {
  basicInfo = 'basic_info',
  clusterInfo = 'cluster_info',
  users = 'users',
  overview = 'overview',
}

export function EngagementTabView({ engagement }: EngagementTabViewProps) {
  const {
    formOptions,
    engagementFormState,
    updateEngagementFormField,
  } = useEngagements();
  const { pathname } = useLocation();
  const history = useHistory();
  const handleTabSelect = (_, tabIndex) => {
    history.push(
      `/app/engagements/${engagementFormState.customer_name}/${engagementFormState.project_name}/${tabIndex}`
    );
  };

  function getActiveKey() {
    if (pathname.includes('overview')) {
      return TabNames.overview;
    } else if (pathname.includes('basic_info')) {
      return TabNames.basicInfo;
    } else if (pathname.includes('cluster_info')) {
      return TabNames.clusterInfo;
    } else if (pathname.includes('users')) {
      return TabNames.users;
    }
    return TabNames.overview;
  }

  return (
    <Tabs isBox activeKey={getActiveKey()} onSelect={handleTabSelect}>
      <Tab title="Overview" eventKey={TabNames.overview} id="overview">
        <TabContentWrapper>
          <EngagementOverview engagement={engagementFormState} />
        </TabContentWrapper>
      </Tab>
      <Tab
        title="Basic Information"
        eventKey={TabNames.basicInfo}
        id="basic_information"
      >
        <TabContentWrapper>
          <EditPaneWrapper engagement={engagementFormState}>
            <BasicInformation
              formOptions={formOptions}
              engagement={engagementFormState}
              onChange={updateEngagementFormField}
            />
            <PointOfContact
              engagement={engagementFormState}
              onChange={updateEngagementFormField}
            />
          </EditPaneWrapper>
        </TabContentWrapper>
      </Tab>
      <Tab title="Users" eventKey={TabNames.users} id="users">
        <TabContentWrapper>
          <EditPaneWrapper engagement={engagementFormState}>
            <ClusterUsers
              formOptions={formOptions}
              engagement={engagementFormState}
              onChange={updateEngagementFormField}
            />
          </EditPaneWrapper>
        </TabContentWrapper>
      </Tab>
      <Tab
        title="Cluster Information"
        eventKey={TabNames.clusterInfo}
        id="cluster_information"
      >
        <TabContentWrapper>
          <EditPaneWrapper engagement={engagementFormState}>
            <ClusterInformation
              formOptions={formOptions}
              engagement={engagementFormState}
              onChange={updateEngagementFormField}
            />
          </EditPaneWrapper>
        </TabContentWrapper>
      </Tab>
    </Tabs>
  );
}

function TabContentWrapper(props: { children: any }) {
  return <div style={{ paddingTop: 20 }}>{props.children}</div>;
}
