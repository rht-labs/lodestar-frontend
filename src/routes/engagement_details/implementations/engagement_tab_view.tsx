import React from 'react';
import { EngagementViewProps } from '..';
import { Tabs, Tab } from '@patternfly/react-core';
import { useEngagements } from '../../../context/engagement_context/engagement_hook';
import { ClusterUsers } from '../tab_views/users';
import { EditPaneWrapper } from '../../../components/edit_pane_wrapper/edit_pane_wrapper';
import { useLocation, useHistory } from 'react-router';
import { EngagementOverviewTab } from '../tab_views/overview';
import { HostingEnvironmentTab } from '../tab_views/hosting_environment';
interface EngagementTabViewProps extends EngagementViewProps {}

enum TabNames {
  overview = 'overview',
  users = 'users',
  hostingEnvironment = 'hosting_environment',
}

export function EngagementTabView({ engagement }: EngagementTabViewProps) {
  const {
    formOptions,
    engagementFormState,
    updateEngagementFormField,
    saveEngagement,
  } = useEngagements();
  const { pathname } = useLocation();
  const history = useHistory();
  const handleTabSelect = (_, tabIndex) => {
    history.push(
      `/app/engagements/${engagementFormState.customer_name}/${engagementFormState.project_name}/${tabIndex}`
    );
  };

  function getActiveKey() {
    const activeTab = Object.keys(TabNames).find(e => {
      return pathname.includes(TabNames[e]);
    });
    if (activeTab) {
      return TabNames[activeTab];
    }
    return TabNames.overview;
  }

  return (
    <Tabs isBox activeKey={getActiveKey()} onSelect={handleTabSelect}>
      <Tab title="Overview" eventKey={TabNames.overview} id="overview">
        <TabContentWrapper>
          <EngagementOverviewTab
            onSave={saveEngagement}
            formOptions={formOptions}
            onChange={updateEngagementFormField}
            engagement={engagementFormState}
          />
        </TabContentWrapper>
      </Tab>
      <Tab title="Users" eventKey={TabNames.users} id="users">
        <TabContentWrapper>
          <EditPaneWrapper engagement={engagementFormState}>
            <ClusterUsers
              formOptions={formOptions}
              engagement={engagementFormState}
              onChange={updateEngagementFormField}
            />
          </EditPaneWrapper>
        </TabContentWrapper>
      </Tab>
      <Tab
        title="Hosting Environment"
        eventKey={TabNames.hostingEnvironment}
        id={'hosting_environment'}
      >
        <TabContentWrapper>
          <EditPaneWrapper engagement={engagementFormState}>
            <HostingEnvironmentTab
              onSave={saveEngagement}
              formOptions={formOptions}
              engagement={engagement}
              onChange={updateEngagementFormField}
            />
          </EditPaneWrapper>
        </TabContentWrapper>
      </Tab>
    </Tabs>
  );
}

function TabContentWrapper(props: { children: any }) {
  return <div style={{ paddingTop: 20 }}>{props.children}</div>;
}
