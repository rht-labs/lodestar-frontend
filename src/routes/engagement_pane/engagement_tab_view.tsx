import React, { useState } from 'react';
import { EngagementViewProps } from '.';
import { Loading } from './Loading';
import { Tabs, Tab } from '@patternfly/react-core';

interface EngagementTabViewProps extends EngagementViewProps {}

export function EngagementTabView({ engagement }: EngagementTabViewProps) {
  const [currentTab, setCurrentTab] = useState(0);
  const handleTabSelect = (e, tabIndex) => setCurrentTab(tabIndex);
  if (!engagement) {
    return <Loading />;
  }
  return (
    <Tabs activeKey={currentTab} onSelect={handleTabSelect}>
      <Tab title="Overview" eventKey={0}>
        Hello
      </Tab>
      <Tab title="Basic Information" eventKey={1}></Tab>
      <Tab title="Users" eventKey={2}></Tab>
      <Tab title="Cluster Information" eventKey={3}></Tab>
    </Tabs>
  );
}
