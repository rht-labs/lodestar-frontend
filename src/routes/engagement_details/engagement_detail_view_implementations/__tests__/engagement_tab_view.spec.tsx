import React from 'react';
import { render } from '@testing-library/react';
import { EngagementTabView } from '../engagement_tab_view';
import { TestStateWrapper } from '../../../../common/test_state_wrapper';
describe('Engagement Tab View', () => {
  test('should match snapshot', () => {
    expect(
      render(
        <TestStateWrapper>
          <EngagementTabView />
        </TestStateWrapper>
      )
    ).toMatchSnapshot();
  });
});
