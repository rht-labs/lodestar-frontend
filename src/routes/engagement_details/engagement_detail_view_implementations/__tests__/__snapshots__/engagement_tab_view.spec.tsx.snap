// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement Tab View should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-tabs pf-m-box"
        data-ouia-component-id="1"
        data-ouia-component-type="PF4/Tabs"
        data-ouia-safe="true"
      >
        <button
          aria-hidden="true"
          aria-label="Scroll left"
          class="pf-c-tabs__scroll-button"
          disabled=""
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              transform=""
            />
          </svg>
        </button>
        <ul
          class="pf-c-tabs__list"
        >
          <li
            class="pf-c-tabs__item pf-m-current"
          >
            <button
              aria-controls="pf-tab-section-overview-overview"
              class="pf-c-tabs__link"
              id="pf-tab-overview-overview"
            >
              Overview
            </button>
          </li>
          <li
            class="pf-c-tabs__item"
          >
            <button
              aria-controls="pf-tab-section-basic_info-basic_information"
              class="pf-c-tabs__link"
              id="pf-tab-basic_info-basic_information"
            >
              Basic Information
            </button>
          </li>
          <li
            class="pf-c-tabs__item"
          >
            <button
              aria-controls="pf-tab-section-users-users"
              class="pf-c-tabs__link"
              id="pf-tab-users-users"
            >
              Users
            </button>
          </li>
          <li
            class="pf-c-tabs__item"
          >
            <button
              aria-controls="pf-tab-section-cluster_info-cluster_information"
              class="pf-c-tabs__link"
              id="pf-tab-cluster_info-cluster_information"
            >
              Cluster Information
            </button>
          </li>
        </ul>
        <button
          aria-hidden="true"
          aria-label="Scroll right"
          class="pf-c-tabs__scroll-button"
          disabled=""
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              transform=""
            />
          </svg>
        </button>
      </div>
      <section
        aria-labelledby="pf-tab-overview-overview"
        class="pf-c-tab-content"
        id="pf-tab-section-overview-overview"
        role="tabpanel"
        tabindex="0"
      >
        <div
          style="padding-top: 20px;"
        >
          <div
            class="pf-c-content"
          >
            <div>
              <div
                class="pf-l-grid pf-m-gutter"
              >
                <div
                  class="pf-l-grid__item pf-m-6-col"
                >
                  <article
                    class="pf-c-card"
                  >
                    <div
                      class="pf-c-card__title"
                    >
                      <h2
                        class="pf-c-title pf-m-xl"
                      >
                        Engagement Summary
                      </h2>
                    </div>
                    <div
                      class="pf-c-card__body"
                    >
                      <div
                        class="pf-l-grid pf-m-gutter"
                      >
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Company
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Project
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Start Date
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              End Date
                            </h6>
                          </div>
                        </div>
                      </div>
                    </div>
                  </article>
                </div>
                <div
                  class="pf-l-grid__item pf-m-6-col"
                >
                  <article
                    class="pf-c-card"
                  >
                    <div
                      class="pf-c-card__title"
                    >
                      <h2
                        class="pf-c-title pf-m-xl"
                      >
                        Points of Contact
                      </h2>
                    </div>
                    <div
                      class="pf-c-card__body"
                    >
                      <div
                        class="pf-l-grid pf-m-gutter"
                      >
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Engagement Lead
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Technical Lead
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-6-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Customer Contact
                            </h6>
                          </div>
                        </div>
                      </div>
                    </div>
                  </article>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col"
                >
                  <article
                    class="pf-c-card"
                  >
                    <div
                      class="pf-c-card__title"
                    >
                      <h2
                        class="pf-c-title pf-m-xl"
                      >
                        Openshift Information
                      </h2>
                    </div>
                    <div
                      class="pf-c-card__body"
                    >
                      <div
                        class="pf-l-grid pf-m-gutter"
                      >
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Cloud Provider
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Openshift Version
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Storage Size
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Cloud Region
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Cloud Provider
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div
                            class="pf-c-content"
                          >
                            <h6
                              class=""
                              data-pf-content="true"
                            >
                              Subdomain
                            </h6>
                          </div>
                        </div>
                        <div
                          class="pf-l-grid__item pf-m-3-col"
                        >
                          <div />
                        </div>
                      </div>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="pf-tab-basic_info-basic_information"
        class="pf-c-tab-content"
        hidden=""
        id="pf-tab-section-basic_info-basic_information"
        role="tabpanel"
        tabindex="0"
      >
        <div
          style="padding-top: 20px;"
        >
          <div />
        </div>
      </section>
      <section
        aria-labelledby="pf-tab-users-users"
        class="pf-c-tab-content"
        hidden=""
        id="pf-tab-section-users-users"
        role="tabpanel"
        tabindex="0"
      >
        <div
          style="padding-top: 20px;"
        >
          <div />
        </div>
      </section>
      <section
        aria-labelledby="pf-tab-cluster_info-cluster_information"
        class="pf-c-tab-content"
        hidden=""
        id="pf-tab-section-cluster_info-cluster_information"
        role="tabpanel"
        tabindex="0"
      >
        <div
          style="padding-top: 20px;"
        >
          <div />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-tabs pf-m-box"
      data-ouia-component-id="1"
      data-ouia-component-type="PF4/Tabs"
      data-ouia-safe="true"
    >
      <button
        aria-hidden="true"
        aria-label="Scroll left"
        class="pf-c-tabs__scroll-button"
        disabled=""
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            transform=""
          />
        </svg>
      </button>
      <ul
        class="pf-c-tabs__list"
      >
        <li
          class="pf-c-tabs__item pf-m-current"
        >
          <button
            aria-controls="pf-tab-section-overview-overview"
            class="pf-c-tabs__link"
            id="pf-tab-overview-overview"
          >
            Overview
          </button>
        </li>
        <li
          class="pf-c-tabs__item"
        >
          <button
            aria-controls="pf-tab-section-basic_info-basic_information"
            class="pf-c-tabs__link"
            id="pf-tab-basic_info-basic_information"
          >
            Basic Information
          </button>
        </li>
        <li
          class="pf-c-tabs__item"
        >
          <button
            aria-controls="pf-tab-section-users-users"
            class="pf-c-tabs__link"
            id="pf-tab-users-users"
          >
            Users
          </button>
        </li>
        <li
          class="pf-c-tabs__item"
        >
          <button
            aria-controls="pf-tab-section-cluster_info-cluster_information"
            class="pf-c-tabs__link"
            id="pf-tab-cluster_info-cluster_information"
          >
            Cluster Information
          </button>
        </li>
      </ul>
      <button
        aria-hidden="true"
        aria-label="Scroll right"
        class="pf-c-tabs__scroll-button"
        disabled=""
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            transform=""
          />
        </svg>
      </button>
    </div>
    <section
      aria-labelledby="pf-tab-overview-overview"
      class="pf-c-tab-content"
      id="pf-tab-section-overview-overview"
      role="tabpanel"
      tabindex="0"
    >
      <div
        style="padding-top: 20px;"
      >
        <div
          class="pf-c-content"
        >
          <div>
            <div
              class="pf-l-grid pf-m-gutter"
            >
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card"
                >
                  <div
                    class="pf-c-card__title"
                  >
                    <h2
                      class="pf-c-title pf-m-xl"
                    >
                      Engagement Summary
                    </h2>
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    <div
                      class="pf-l-grid pf-m-gutter"
                    >
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Company
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Project
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Start Date
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            End Date
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card"
                >
                  <div
                    class="pf-c-card__title"
                  >
                    <h2
                      class="pf-c-title pf-m-xl"
                    >
                      Points of Contact
                    </h2>
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    <div
                      class="pf-l-grid pf-m-gutter"
                    >
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Engagement Lead
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Technical Lead
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-6-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Customer Contact
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-12-col"
              >
                <article
                  class="pf-c-card"
                >
                  <div
                    class="pf-c-card__title"
                  >
                    <h2
                      class="pf-c-title pf-m-xl"
                    >
                      Openshift Information
                    </h2>
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    <div
                      class="pf-l-grid pf-m-gutter"
                    >
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Cloud Provider
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Openshift Version
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Storage Size
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Cloud Region
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Cloud Provider
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div
                          class="pf-c-content"
                        >
                          <h6
                            class=""
                            data-pf-content="true"
                          >
                            Subdomain
                          </h6>
                        </div>
                      </div>
                      <div
                        class="pf-l-grid__item pf-m-3-col"
                      >
                        <div />
                      </div>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      aria-labelledby="pf-tab-basic_info-basic_information"
      class="pf-c-tab-content"
      hidden=""
      id="pf-tab-section-basic_info-basic_information"
      role="tabpanel"
      tabindex="0"
    >
      <div
        style="padding-top: 20px;"
      >
        <div />
      </div>
    </section>
    <section
      aria-labelledby="pf-tab-users-users"
      class="pf-c-tab-content"
      hidden=""
      id="pf-tab-section-users-users"
      role="tabpanel"
      tabindex="0"
    >
      <div
        style="padding-top: 20px;"
      >
        <div />
      </div>
    </section>
    <section
      aria-labelledby="pf-tab-cluster_info-cluster_information"
      class="pf-c-tab-content"
      hidden=""
      id="pf-tab-section-cluster_info-cluster_information"
      role="tabpanel"
      tabindex="0"
    >
      <div
        style="padding-top: 20px;"
      >
        <div />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
